ironfish/src/network/peers/peerManager.ts

```diff
@@ -782,55 +793,54 @@ export class PeerManager {
     }
   }
 
-  start(): void {
-    this.distributePeerListHandle = setInterval(() => this.distributePeerList(), 5000)
-    this.disposePeersHandle = setInterval(() => this.disposePeers(), 2000)
+  async start(): Promise<void> {
+    await Promise.allSettled([
+      (this.requestPeerListHandle = setInterval(() => this.requestPeerList(), 60000)),
+      (this.disposePeersHandle = setInterval(() => this.disposePeers(), 2000)),
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      (this.savePeerAddressesHandle = setInterval(async () => {
+        await this.addressManager.save(this.peers)
```

Comment:
You can drop the async function and just do `void this.addressManager.save(this.peers)`. This will allow you to remove the disabled lint line as well.

Reply:
Why is the peer manager telling the AddressManager when to save? We're interesting the first top level system which doesn't control it's own event loop. So were diverging from other examples in the code base now right?

Reply:
That's fair; I'll find a way to persist currently connected peers without having to directly involve the PeerManager.

Reply:
After looking at this over the weekend, I don't think that there's a clean way to do this other than what is already being done. AddressManager needs to get the peer information somehow to be able to save it to disk and all of that information lives in PeerManager. We can either do it this way or have AddressManager contain a reference to PeerManager somehow, but I'm not sure that makes the code any easier to follow.